rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -535,16 +536,17 @@ class WriterInfo : public ReaderListener
     bool trigger = false;
     mapmutex.lock();
     if (change->kind == ALIVE) {
-      topicNtypes[fqdn].insert(proxyData.typeName());
+      topicNtypes[fqdn].push_back(proxyData.typeName());
       trigger = true;
     } else {
       auto it = topicNtypes.find(fqdn);
-      if (
-        it != topicNtypes.end() &&
-        it->second.find(proxyData.typeName()) != it->second.end())
-      {
-        topicNtypes[fqdn].erase(proxyData.typeName());
-        trigger = true;
+      if (it != topicNtypes.end()) {
+        const auto & loc =
+          std::find(std::begin(it->second), std::end(it->second), proxyData.typeName());
+        if (loc != std::end(it->second)) {
+          topicNtypes[fqdn].erase(loc, loc + 1);
+          trigger = true;
+        }
```

Comment:
Is the else case here an error? Seems to me that if we get an event for deregistering a type that we're not currently tracking it might be a bug?

The one case I could imagine that would be a false positive is that a subscriber comes up and goes down, but this program only receives the latter notification for some reason. Maybe it deserves a warning? I'm not sure. It seems like a place where a real bug could go unnoticed because we're ignoring this atypical case.